source/fundamentals/typescript.txt

```diff
@@ -79,81 +89,13 @@ You can find a code snippet that shows how to specify a type for the ``FindCurso
 
-Limitations
------------
-
-.. _node-driver-typescript-limitations-dot-notation:
-
-The driver cannot infer the type of values with keys containing **dot
-notation**. Dot notation is a property access syntax for navigating BSON objects.
-Click on the tabs to see code snippets that highlight this behavior:
-
-.. tabs::
-
-  .. tab:: Dot Notation
-     :tabid: dot-notation
-
-     The following code snippet does not raise a type error:
-
-     .. literalinclude:: /code-snippets/typescript/dot-notation.ts
-        :language: typescript
-        :linenos:
-        :start-after: start-no-error
-        :end-before: end-no-error
-
-  .. tab:: Nested Objects
-     :tabid: nested-objects
-
-     The following code snippet raises a type error:
-
-     .. literalinclude:: /code-snippets/typescript/dot-notation.ts
-        :language: typescript
-        :linenos:
-        :start-after: start-error
-        :end-before: end-error
-
-     This is the error:
-
-     .. code-block:: text
+To view an example of a mutually recursive type, which is not supported by the
+:ref:`preceding classes <node-mongodb-type-paramaters-any-type>`,
+see the :ref:`<node-driver-limitations-mutual-recursion>` section.
 
-        Type 'string' is not assignable to type 'number'.
-
-Despite the lack of type safety, we still recommend that you use dot notation to
-access nested fields in query and update documents when you use TypeScript. You
-must manually check that your nested field values have your intended type.
-
-.. note:: Reason To Use Dot Notation
-
-   In the MongoDB Query Language, you must match a subdocument exactly
-   when specifying subdocuments in a query. Dot notation allows you to query
-   nested fields without matching subdocuments exactly.
-
-   To show this behavior, lets say you have a collection containing
-   only the following document:
-
-   .. code-block:: json
-
-      { field: { s1: "hi", s2: "bye" } }
-
-   The following query returns no results from this collection, as the value of
-   ``field`` does not exactly match ``{ s1: "hi" }``:
-
-   .. literalinclude:: /code-snippets/typescript/note-on-dot-notation.ts
-      :language: typescript
-      :linenos:
-      :start-after: start-no-doc
-      :end-before: end-no-doc
-
-   The following queries both return your document:
-
-   .. literalinclude:: /code-snippets/typescript/note-on-dot-notation.ts
-      :language: typescript
-      :linenos:
-      :start-after: start-doc
-      :end-before: end-doc
+.. _node-driver-limitations:
 
-   The syntax of the query that does not use dot notation is cumbersome and hard
-   to understand, and may not be worth the type safety obtained from
-   avoiding dot notation.
+Limitations For Driver Version {+version+}
+----------------------------------
 
-For more information on dot notation, see :manual:`the MongoDB Manual </core/document/#dot-notation>`.
+.. include:: includes/limitations/{+version+}.rst
```

Comment:
I've moved the entire section into the limitations file. Agree it makes more sense to have the entire section within the file (weird to have everything but the heading there), and a "limitations file" should not be a requirement for a successful build (hopefully in the future a limitations file is no longer needed).

Reply:
Unfortunately, the header is placed at an `h3` level if drawn from the included file. I've added the header back to the original TypeScript file. I agree this will require the addition of a new limitations file when the driver is updated to v4.4.